oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/web/DefaultBearerTokenResolver.java

```diff
@@ -81,47 +85,65 @@ public void setAllowFormEncodedBodyParameter(boolean allowFormEncodedBodyParamet
 	 *
 	 * @param allowUriQueryParameter if the URI query parameter is supported
 	 */
-	public void setAllowUriQueryParameter(boolean allowUriQueryParameter) {
+	public void setAllowUriQueryParameter(final boolean allowUriQueryParameter) {
 		this.allowUriQueryParameter = allowUriQueryParameter;
 	}
 
-	private static String resolveFromAuthorizationHeader(HttpServletRequest request) {
-		String authorization = request.getHeader(HttpHeaders.AUTHORIZATION);
-		if (StringUtils.hasText(authorization) && authorization.startsWith("Bearer")) {
-			Matcher matcher = authorizationPattern.matcher(authorization);
+	private void validate(final HttpServletRequest request, final String authorizationHeaderToken) {
+		if (hasMultipleAccessTokens(request, authorizationHeaderToken)) {
+			throw new OAuth2AuthenticationException(BEARER_TOKEN_ERROR_MULTIPLE);
+		}
+		if (this.allowFormEncodedBodyParameter == true) {
+			if (this.allowUriQueryParameter == false && isTokenInUriQueryParameter(request)) {
+				throw new OAuth2AuthenticationException(BEARER_TOKEN_ERROR_MALFORMED);
+			}
+		}
+	}
 
+	private String resolveFromAuthorizationHeader(final HttpServletRequest request) {
+		final String authorization = request.getHeader(HttpHeaders.AUTHORIZATION);
+		if (authorization != null && authorization.startsWith("Bearer")) {
+			final Matcher matcher = AUTHORIZATION_PATTERN.matcher(authorization);
 			if (!matcher.matches()) {
-				BearerTokenError error = new BearerTokenError(BearerTokenErrorCodes.INVALID_TOKEN,
-						HttpStatus.UNAUTHORIZED,
-						"Bearer token is malformed",
-						"https://tools.ietf.org/html/rfc6750#section-3.1");
-				throw new OAuth2AuthenticationException(error);
+				throw new OAuth2AuthenticationException(BEARER_TOKEN_ERROR_MALFORMED);
 			}
-
 			return matcher.group("token");
 		}
 		return null;
 	}
 
-	private static String resolveFromRequestParameters(HttpServletRequest request) {
-		String[] values = request.getParameterValues("access_token");
-		if (values == null || values.length == 0)  {
+	private String resolveFromRequestParameters(final HttpServletRequest request) {
+		// By this point there is either zero or one value
+		final String value = request.getParameter(ACCESS_TOKEN_PARAM_NAME);
+		if (value != null && isParameterTokenSupportedForRequest(request)) {
+			final Matcher matcher = JWT_PATTERN.matcher(value);
+			if (!matcher.matches()) {
+				throw new OAuth2AuthenticationException(BEARER_TOKEN_ERROR_MALFORMED);
+			}
+			return value;
+		} else {
 			return null;
 		}
+	}
 
-		if (values.length == 1) {
-			return values[0];
-		}
+	private boolean isParameterTokenSupportedForRequest(final HttpServletRequest request) {
+		return this.allowUriQueryParameter || (this.allowFormEncodedBodyParameter
+				&& MediaType.APPLICATION_FORM_URLENCODED.equals(MediaType.valueOf(request.getContentType()))
+				&& !RequestMethod.GET.equals(RequestMethod.valueOf(request.getMethod())));
```

Comment:
If `Content-Type` has been presented then it isn't a `GET` request so this line is needless, right?